src/overlays/actors/ovl_Dm_Char09/z_dm_char09.c

```diff
@@ -32,28 +30,197 @@ const ActorInit Dm_Char09_InitVars = {
+    SkelAnime_Init(play, &this->skelAnime, &object_bee_Skel_001398, &object_bee_Anim_00005C, this->jointTable,
+                   this->morphTable, OBJECT_BEE_LIMB_MAX);
+    DmChar09_ChangeAnimation(&this->skelAnime, sAnimationInfo, 0);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->unk_228 = Rand_ZeroOne() * 65535.0f;
+    this->unk_22A = Rand_ZeroOne() * 65535.0f;
+    this->actionFunc = DmChar09_DoNothing;
+}
+
+void DmChar09_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80AB1FA0(DmChar09* this, s32 arg1) {
+    Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->unk_224[arg1]);
+}
+
+void func_80AB1FDC(DmChar09* this, PlayState* play) {
+    Actor* thisx = &this->actor;
+    Vec3f sp58;
+    f32 sp54;
+    f32 phi_fa0;
+    f32 phi_fv0;
+    Vec3f sp40;
+    s32 phi_a1;
+    Vec3s* temp_v1;
+
+    Math_Vec3f_Copy(&sp40, &thisx->world.pos);
+    Math_Vec3s_ToVec3f(&sp58, this->unk_224 + this->unk_21C + this->unk_220);
+    Math_Vec3f_Diff(&sp58, &thisx->world.pos, &thisx->velocity);
+    sp54 = Math3D_Vec3fMagnitude(&thisx->velocity);
+    if ((sp54 < (this->speed * 8.0f)) && (this->speed > 2.0f)) {
+        phi_fv0 = ((this->speed - 2.0f) * 0.1f) + 2.0f;
+        phi_fa0 = this->speed * 0.03f;
+    } else {
+        phi_fv0 = this->speed;
+        phi_fa0 = this->speed * 0.16f;
+    }
+    Math_StepToF(&thisx->speedXZ, phi_fv0, phi_fa0);
+    if ((thisx->speedXZ + 0.05f) < sp54) {
+        Math_Vec3f_Scale(&thisx->velocity, thisx->speedXZ / sp54);
+        thisx->world.pos.x += thisx->velocity.x;
+        thisx->world.pos.y += thisx->velocity.y;
+        thisx->world.pos.z += thisx->velocity.z;
+    } else {
+        this->unk_21C += this->unk_220;
+        thisx->speedXZ *= 0.4f;
+        phi_a1 = true;
+        if (((this->unk_21C >= this->unk_218) && (this->unk_220 > 0)) ||
+            ((this->unk_21C <= 0) && (this->unk_220 < 0))) {
+            temp_v1 = this->unk_224 + this->unk_218;
+            if (((this->unk_224->x == temp_v1->x) && (this->unk_224->y == temp_v1->y)) &&
+                (this->unk_224->z == temp_v1->z)) {
+                this->unk_21C = 0;
+                this->unk_220 = 1;
+            } else {
+                phi_a1 = false;
+                this->actionFunc = func_80AB2268;
+            }
+        }
+        if (phi_a1) {
+            func_80AB1FA0(this, this->unk_21C);
+        }
+    }
+    Math_SmoothStepToS(&thisx->world.rot.y, Math_Vec3f_Yaw(&thisx->world.pos, &sp40) + 0x7FFF, 1, 0x7D0, 0);
+    thisx->shape.rot.y = thisx->world.rot.y;
+}
+
+void DmChar09_DoNothing(DmChar09* this, PlayState* play) {
+}
+
+void func_80AB2268(DmChar09* this, PlayState* play) {
+    Path* path;
+    s32 pad;
+    s32 i;
+    s32 actionIndex;
+    s32 action = 0;
+    s32 pathnum;
+    u8 csIndex = 0;
+
+    if (!DMCHAR09_GET_F(&this->actor)) {
+        if (play->csCtx.currentCsIndex == 1) {
+            csIndex = 1;
+        }
+    } else if (play->csCtx.currentCsIndex == 0) {
+        csIndex = 1;
+    }
```

Comment:
```suggestion
    u8 csIndex = false;

    if (!DMCHAR09_GET_F(&this->actor)) {
        if (play->csCtx.currentCsIndex == 1) {
            csIndex = true;
        }
    } else if (play->csCtx.currentCsIndex == 0) {
        csIndex = true;
    }
```
Also `csIndex` does not look like an index, but a boolean of some kind. I'm not sure on a good name, so probably prefer just something generic like temp for now.

Reply:
Are you sure about this? because even if 0 and 1 are the only two uses in this file, if you look at the repo, currentCsIndex can also be 2, 3 and even 4, so if this variable represents csIndex, it shouldn't be a boolean.

Reply:
I'm not referring to `currentCsIndex`, but `csIndex` which is just a stack variable in this function that seems to act like a boolean.

Reply:
now i'm seeing it from your perspective, and yes, i agree. csIndex goes back to temp until we know proper behaviour, thank you